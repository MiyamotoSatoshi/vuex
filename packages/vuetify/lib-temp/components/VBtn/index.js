import VBtn from './VBtn';
export { VBtn };
export default VBtn;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WQnRuL2luZGV4LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sSUFBSSxNQUFNLFFBQVEsQ0FBQTtBQUV6QixPQUFPLEVBQUUsSUFBSSxFQUFFLENBQUE7QUFDZixlQUFlLElBQUksQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBWQnRuIGZyb20gJy4vVkJ0bidcblxuZXhwb3J0IHsgVkJ0biB9XG5leHBvcnQgZGVmYXVsdCBWQnRuXG4iXX0=