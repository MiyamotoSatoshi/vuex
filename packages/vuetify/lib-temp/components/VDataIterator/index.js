import VDataIterator from './VDataIterator';
export { VDataIterator };
export default VDataIterator;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WRGF0YUl0ZXJhdG9yL2luZGV4LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sYUFBYSxNQUFNLGlCQUFpQixDQUFBO0FBRTNDLE9BQU8sRUFBRSxhQUFhLEVBQUUsQ0FBQTtBQUN4QixlQUFlLGFBQWEsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBWRGF0YUl0ZXJhdG9yIGZyb20gJy4vVkRhdGFJdGVyYXRvcidcblxuZXhwb3J0IHsgVkRhdGFJdGVyYXRvciB9XG5leHBvcnQgZGVmYXVsdCBWRGF0YUl0ZXJhdG9yXG4iXX0=