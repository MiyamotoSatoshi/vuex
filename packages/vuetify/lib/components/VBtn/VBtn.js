var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

// Styles
import '../../../src/stylus/components/_buttons.styl';
import mixins from '../../util/mixins';
// Components
import VProgressCircular from '../VProgressCircular';
// Mixins
import Colorable from '../../mixins/colorable';
import { factory as GroupableFactory } from '../../mixins/groupable';
import Positionable from '../../mixins/positionable';
import Routable from '../../mixins/routable';
import Themeable from '../../mixins/themeable';
import { factory as ToggleableFactory } from '../../mixins/toggleable';
export default mixins(Colorable, Routable, Positionable, Themeable, GroupableFactory('btnToggle'), ToggleableFactory('inputValue')
/* @vue/component */
).extend({
    name: 'v-btn',
    props: {
        activeClass: {
            type: String,
            default: 'v-btn--active'
        },
        block: Boolean,
        depressed: Boolean,
        fab: Boolean,
        flat: Boolean,
        icon: Boolean,
        large: Boolean,
        loading: Boolean,
        outline: Boolean,
        ripple: {
            type: [Boolean, Object],
            default: null
        },
        round: Boolean,
        small: Boolean,
        tag: {
            type: String,
            default: 'button'
        },
        type: {
            type: String,
            default: 'button'
        },
        value: null
    },
    computed: {
        classes: function classes() {
            var _extends2;

            return _extends((_extends2 = {
                'v-btn': true
            }, _defineProperty(_extends2, this.activeClass, this.isActive), _defineProperty(_extends2, 'v-btn--absolute', this.absolute), _defineProperty(_extends2, 'v-btn--block', this.block), _defineProperty(_extends2, 'v-btn--bottom', this.bottom), _defineProperty(_extends2, 'v-btn--disabled', this.disabled), _defineProperty(_extends2, 'v-btn--flat', this.flat), _defineProperty(_extends2, 'v-btn--floating', this.fab), _defineProperty(_extends2, 'v-btn--fixed', this.fixed), _defineProperty(_extends2, 'v-btn--icon', this.icon), _defineProperty(_extends2, 'v-btn--large', this.large), _defineProperty(_extends2, 'v-btn--left', this.left), _defineProperty(_extends2, 'v-btn--loader', this.loading), _defineProperty(_extends2, 'v-btn--outline', this.outline), _defineProperty(_extends2, 'v-btn--depressed', this.depressed && !this.flat || this.outline), _defineProperty(_extends2, 'v-btn--right', this.right), _defineProperty(_extends2, 'v-btn--round', this.round), _defineProperty(_extends2, 'v-btn--router', this.to), _defineProperty(_extends2, 'v-btn--small', this.small), _defineProperty(_extends2, 'v-btn--top', this.top), _extends2), this.themeClasses);
        },
        computedRipple: function computedRipple() {
            var defaultRipple = this.icon || this.fab ? { circle: true } : true;
            if (this.disabled) return false;else return this.ripple !== null ? this.ripple : defaultRipple;
        }
    },
    methods: {
        // Prevent focus to match md spec
        click: function click(e) {
            !this.fab && e.detail && this.$el.blur();
            this.$emit('click', e);
            this.btnToggle && this.toggle();
        },
        genContent: function genContent() {
            return this.$createElement('div', { 'class': 'v-btn__content' }, [this.$slots.default]);
        },
        genLoader: function genLoader() {
            var children = [];
            if (!this.$slots.loader) {
                children.push(this.$createElement(VProgressCircular, {
                    props: {
                        indeterminate: true,
                        size: 23,
                        width: 2
                    }
                }));
            } else {
                children.push(this.$slots.loader);
            }
            return this.$createElement('span', { 'class': 'v-btn__loading' }, children);
        }
    },
    render: function render(h) {
        var setColor = !this.outline && !this.flat ? this.setBackgroundColor : this.setTextColor;

        var _generateRouteLink = this.generateRouteLink(this.classes),
            tag = _generateRouteLink.tag,
            data = _generateRouteLink.data;

        var children = [this.genContent()];
        tag === 'button' && (data.attrs.type = this.type);
        this.loading && children.push(this.genLoader());
        data.attrs.value = ['string', 'number'].includes(_typeof(this.value)) ? this.value : JSON.stringify(this.value);
        return h(tag, setColor(this.color, data), children);
    }
});
//# sourceMappingURL=VBtn.js.map