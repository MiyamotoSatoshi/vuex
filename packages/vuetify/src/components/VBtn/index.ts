export { default as VBtn } from './VBtn'
