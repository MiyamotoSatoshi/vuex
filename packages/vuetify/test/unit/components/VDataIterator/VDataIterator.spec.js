import Vue from 'vue'
import { test } from '@/test'
import VDataIterator from '@/components/VDataIterator'
import VBtn from '@/components/VBtn'

test.skip('VDataIterator.js', ({ mount, compileToFunctions }) => {
  function dataIteratorTestData () {
    return {
      propsData: {
        pagination: {
          descending: false,
          sortBy: 'col1',
          rowsPerPage: 5,
          page: 1
        },
        items: [
          { other: 1, col1: 'foo', col2: 'a', col3: 1 },
          { other: 2, col1: null, col2: 'b', col3: 2 },
          { other: 3, col1: undefined, col2: 'c', col3: 3 }
        ]
      }
    }
  }

  it('should match a snapshot - no matching records', () => {
    const data = dataIteratorTestData()
    data.propsData.search = "asdf"
    const wrapper = mount(VDataIterator, data)

    expect(wrapper.html()).toMatchSnapshot()

    const content = wrapper.find('.v-data-iterator div div')[0]
    expect(content.element.textContent).toBe('No matching records found')

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should match a snapshot - hideActions and no footer slot', () => {
    const data = dataIteratorTestData()
    data.propsData.hideActions = true
    const wrapper = mount(VDataIterator, data)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match a snapshot - footer slot', () => {
    const data = dataIteratorTestData()
    data.slots = {
      footer: [compileToFunctions('<span>footer</span>')],
    }
    const wrapper = mount(VDataIterator, data)

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should match a snapshot - no data', () => {
    const data = dataIteratorTestData()
    data.propsData.items = []
    const wrapper = mount(VDataIterator, data)

    expect(wrapper.html()).toMatchSnapshot()

    const content = wrapper.find('.v-data-iterator div div')[0]
    expect(content.element.textContent).toBe('No data available')

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should match a snapshot - with data', async () => {
    const data = dataIteratorTestData()

    const vm = new Vue()
    const item = props => vm.$createElement('div', [props.item.col2])
    const component = Vue.component('test', {
      render (h) {
        return h(VDataIterator, {
          props: {
            'content-tag': 'span',
            ...data.propsData
          },
          scopedSlots: {
            item
          }
        })
      }
    })

    const wrapper = mount(component)

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.vm.$vuetify.rtl = true
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.v-data-iterator__actions__range-controls')[0].html()).toMatchSnapshot()
    wrapper.vm.$vuetify.rtl = undefined

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should pass attrs, class and props to content', () => {
    const data = dataIteratorTestData()

    const vm = new Vue()
    const item = props => vm.$createElement('div', [props.item.col2])
    const component = Vue.component('test', {
      render (h) {
        return h(VDataIterator, {
          props: {
            'content-tag': 'button', // TODO: use v-btn so we can test content-props
            ...data.propsData,
            // 'content-props': { block: true }, // TODO: enable when migrating to vue-test-utils
            'content-class': 'test__class'
          },
          attrs: {
            id: "testButtonId"
          },
          scopedSlots: {
            item
          }
        })
      }
    })

    const wrapper = mount(component)

    const mainDiv = wrapper.find('.v-data-iterator')[0]
    expect(mainDiv.hasAttribute('id')).toBe(false)

    var button = mainDiv.find('button')[0]
    expect(button.getAttribute('id')).toBe('testButtonId')
    // expect(button.hasClass('v-btn--block')).toBe(true) // TODO: enable when migrating to vue-test-utils
    expect(button.hasClass('test__class')).toBe(true)

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should not filter items if search is empty', async () => {
    const data = dataIteratorTestData()
    data.propsData.search = '    '
    const wrapper = mount(VDataIterator, data)

    expect(wrapper.instance().filteredItems).toHaveLength(data.propsData.items.length)

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render header slot', async () => {
    const data = dataIteratorTestData()
    data.slots = {
      footer: [compileToFunctions('<span class="header">header</span>')],
    }

    const wrapper = mount(VDataIterator, data)

    expect(wrapper.find('span.header').length).toBe(1)
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should reset to page 1 when data is smaller than current page', async () => {
    const data = dataIteratorTestData()
    data.propsData.pagination.rowsPerPage = 1
    data.propsData.pagination.page = 2

    const wrapper = mount(VDataIterator, data)

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.v-data-iterator__actions__pagination')[0].text()).toBe('2-2 of 3')

    wrapper.setProps({ items: [] })

    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.v-data-iterator__actions__pagination')[0].text()).toBe('â€“')
    expect(wrapper.find('[aria-label="Previous page"]')[0].hasClass('v-btn--disabled')).toBe(false)
    expect(wrapper.find('[aria-label="Next page"]')[0].hasClass('v-btn--disabled')).toBe(true)

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })
})
