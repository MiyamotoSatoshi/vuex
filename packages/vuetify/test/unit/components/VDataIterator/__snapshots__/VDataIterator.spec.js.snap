// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDataIterator.js should match a snapshot - footer slot 1`] = `

<div class="v-data-iterator theme--light">
  <div>
  </div>
  <div>
    <span>
      footer
    </span>
    <div class="v-data-iterator__actions">
      <div class="v-data-iterator__actions__select">
        Items per page:
        <div role="combobox"
             class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">
                    5
                  </div>
                  <input aria-label="Items per page:"
                         readonly="readonly"
                         type="text"
                         autocomplete="on"
                         aria-readonly="false"
                  >
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append">
                    <i aria-hidden="true"
                       class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div class="v-menu">
                <div class="v-menu__content theme--light v-menu__content--auto "
                     style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="v-select-list v-card theme--light">
                    <div role="list"
                         class="v-list theme--light"
                    >
                      <div role="listitem"
                           class="primary--text"
                      >
                        <a class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              5
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              10
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              25
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              All
                            </div>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-iterator__actions__range-controls">
        <div class="v-data-iterator__actions__pagination">
          1-3 of 3
        </div>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Previous page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Next page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDataIterator.js should match a snapshot - hideActions and no footer slot 1`] = `

<div class="v-data-iterator theme--light">
  <div>
  </div>
</div>

`;

exports[`VDataIterator.js should match a snapshot - no data 1`] = `

<div class="v-data-iterator theme--light">
  <div>
    <div class="text-xs-center"
         style="width: 100%;"
    >
      No data available
    </div>
  </div>
  <div>
    <div class="v-data-iterator__actions">
      <div class="v-data-iterator__actions__select">
        Items per page:
        <div role="combobox"
             class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">
                    5
                  </div>
                  <input aria-label="Items per page:"
                         readonly="readonly"
                         type="text"
                         autocomplete="on"
                         aria-readonly="false"
                  >
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append">
                    <i aria-hidden="true"
                       class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div class="v-menu">
                <div class="v-menu__content theme--light v-menu__content--auto "
                     style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="v-select-list v-card theme--light">
                    <div role="list"
                         class="v-list theme--light"
                    >
                      <div role="listitem"
                           class="primary--text"
                      >
                        <a class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              5
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              10
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              25
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              All
                            </div>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-iterator__actions__range-controls">
        <div class="v-data-iterator__actions__pagination">
          â€“
        </div>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Previous page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Next page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDataIterator.js should match a snapshot - no matching records 1`] = `

<div class="v-data-iterator theme--light">
  <div>
    <div class="text-xs-center"
         style="width: 100%;"
    >
      No matching records found
    </div>
  </div>
  <div>
    <div class="v-data-iterator__actions">
      <div class="v-data-iterator__actions__select">
        Items per page:
        <div role="combobox"
             class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">
                    5
                  </div>
                  <input aria-label="Items per page:"
                         readonly="readonly"
                         type="text"
                         autocomplete="on"
                         aria-readonly="false"
                  >
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append">
                    <i aria-hidden="true"
                       class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div class="v-menu">
                <div class="v-menu__content theme--light v-menu__content--auto "
                     style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="v-select-list v-card theme--light">
                    <div role="list"
                         class="v-list theme--light"
                    >
                      <div role="listitem"
                           class="primary--text"
                      >
                        <a class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              5
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              10
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              25
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              All
                            </div>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-iterator__actions__range-controls">
        <div class="v-data-iterator__actions__pagination">
          â€“
        </div>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Previous page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Next page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDataIterator.js should match a snapshot - with data 1`] = `

<div class="v-data-iterator theme--light">
  <span>
    <div>
      b
    </div>
    <div>
      c
    </div>
    <div>
      a
    </div>
  </span>
  <div>
    <div class="v-data-iterator__actions">
      <div class="v-data-iterator__actions__select">
        Items per page:
        <div role="combobox"
             class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">
                    5
                  </div>
                  <input aria-label="Items per page:"
                         readonly="readonly"
                         type="text"
                         autocomplete="on"
                         aria-readonly="false"
                  >
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append">
                    <i aria-hidden="true"
                       class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div class="v-menu">
                <div class="v-menu__content theme--light v-menu__content--auto "
                     style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="v-select-list v-card theme--light">
                    <div role="list"
                         class="v-list theme--light"
                    >
                      <div role="listitem"
                           class="primary--text"
                      >
                        <a class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              5
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              10
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              25
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              All
                            </div>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-iterator__actions__range-controls">
        <div class="v-data-iterator__actions__pagination">
          1-3 of 3
        </div>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Previous page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button disabled="disabled"
                type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Next page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDataIterator.js should match a snapshot - with data 2`] = `

<div class="v-data-iterator__actions__range-controls">
  <div class="v-data-iterator__actions__pagination">
    1-3 of 3
  </div>
  <button disabled="disabled"
          type="button"
          class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
          aria-label="Previous page"
  >
    <div class="v-btn__content">
      <i aria-hidden="true"
         class="v-icon material-icons theme--light"
      >
        chevron_right
      </i>
    </div>
  </button>
  <button disabled="disabled"
          type="button"
          class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
          aria-label="Next page"
  >
    <div class="v-btn__content">
      <i aria-hidden="true"
         class="v-icon material-icons theme--light"
      >
        chevron_left
      </i>
    </div>
  </button>
</div>

`;

exports[`VDataIterator.js should reset to page 1 when data is smaller than current page 1`] = `

<div class="v-data-iterator theme--light">
  <div>
  </div>
  <div>
    <div class="v-data-iterator__actions">
      <div class="v-data-iterator__actions__select">
        Items per page:
        <div role="combobox"
             class="v-input v-text-field v-select v-input--hide-details theme--light"
        >
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <input aria-label="Items per page:"
                         readonly="readonly"
                         type="text"
                         autocomplete="on"
                         aria-readonly="false"
                  >
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append">
                    <i aria-hidden="true"
                       class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div class="v-menu">
                <div class="v-menu__content theme--light v-menu__content--auto "
                     style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="v-select-list v-card theme--light">
                    <div role="list"
                         class="v-list theme--light"
                    >
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              5
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              10
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              25
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              All
                            </div>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-iterator__actions__range-controls">
        <div class="v-data-iterator__actions__pagination">
          2-2 of 3
        </div>
        <button type="button"
                class="v-btn v-btn--flat v-btn--icon theme--light"
                aria-label="Previous page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button type="button"
                class="v-btn v-btn--flat v-btn--icon theme--light"
                aria-label="Next page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;

exports[`VDataIterator.js should reset to page 1 when data is smaller than current page 2`] = `

<div class="v-data-iterator theme--light">
  <div>
    <div class="text-xs-center"
         style="width: 100%;"
    >
      No data available
    </div>
  </div>
  <div>
    <div class="v-data-iterator__actions">
      <div class="v-data-iterator__actions__select">
        Items per page:
        <div role="combobox"
             class="v-input v-text-field v-select v-input--hide-details theme--light"
        >
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <input aria-label="Items per page:"
                         readonly="readonly"
                         type="text"
                         autocomplete="on"
                         aria-readonly="false"
                  >
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append">
                    <i aria-hidden="true"
                       class="v-icon material-icons theme--light"
                    >
                      arrow_drop_down
                    </i>
                  </div>
                </div>
              </div>
              <div class="v-menu">
                <div class="v-menu__content theme--light v-menu__content--auto "
                     style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="v-select-list v-card theme--light">
                    <div role="list"
                         class="v-list theme--light"
                    >
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              5
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              10
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              25
                            </div>
                          </div>
                        </a>
                      </div>
                      <div role="listitem">
                        <a class="v-list__tile v-list__tile--link theme--light">
                          <div class="v-list__tile__content">
                            <div class="v-list__tile__title">
                              All
                            </div>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-iterator__actions__range-controls">
        <div class="v-data-iterator__actions__pagination">
          â€“
        </div>
        <button type="button"
                class="v-btn v-btn--flat v-btn--icon theme--light"
                aria-label="Previous page"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_left
            </i>
          </div>
        </button>
        <button type="button"
                class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light"
                aria-label="Next page"
                disabled="disabled"
        >
          <div class="v-btn__content">
            <i aria-hidden="true"
               class="v-icon material-icons theme--light"
            >
              chevron_right
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;
